### ISSUE TITLE
Feature request - disallowing shared runners on group level

### ISSUE THREAD

**Issue by torinthiel** [gitlabhq/terraform-provider-gitlab#580]
The gitlab_project resource has an attribute `shared_runners_enabled`, which corresponds to similar setting in gitlab.
However this setting exists as well on the group level, along with another one that blocks override on subgroups/projects.

It would be nice to be able to set this on the group level as well.

**Comment by armsnyder** [comment]
This is a request to add the `shared_runners_enabled` attribute to the `gitlab_group` resource.

**Comment by GertVil** [comment]
I was curious and took a look into this issue, but at a first glance, this does not look as straightforward. Isn't it an issue that the Create Group API does not support setting the `shared_runners_setting` on creation? According to the documentation, it is only present on the Update Group.

**Comment by armsnyder** [comment]
Yep, there are some GitLab APIs like that. It's less straightforward, but we do have a way of handling it. In the provider "create" function...
1. Call the create API
2. Set the resource Id (`d.SetId(...)`)
3. Call the update API
4. return

**Comment by teekhandekar** [comment]
@torinthiel Can I work on this ?

**Comment by timofurrer** [comment]
@teekhandekar Yes, feel free to open a PR for that. @armsnyder has pretty much laid out the implementation steps already.

In addition, can you implement all the *shared runners* fields. Looking through the docs I can at least see these:

* Available on create:
    * [`shared_runners_minutes_limit`](https://docs.gitlab.com/ee/api/groups.html#:~:text=branch%20protection%20setting.-,shared_runners_minutes_limit,be%20nil%20(default%3B%20inherit%20system%20default)%2C%200%20(unlimited)%2C%20or%20%3E%200.,-extra_shared_runners_minutes_limit)
    * [`extra_shared_runners_minutes_limit`](https://docs.gitlab.com/ee/api/groups.html#:~:text=unlimited)%2C%20or%20%3E%200.-,extra_shared_runners_minutes_limit,by%20administrators%20only.%20Additional%20CI/CD%20minutes%20for%20this%20group.,-Options%20for%20default_branch_protection)
* Available on update:
    * [`shared_runners_setting`](https://docs.gitlab.com/ee/api/groups.html#:~:text=shared_runners_setting,runners%20for%20a%20group%E2%80%99s%20subgroups%20and%20projects.) also see [the options for that](https://docs.gitlab.com/ee/api/groups.html#options-for-shared_runners_setting)
* Available on read:
    * [`shared_runners_enabled`](https://docs.gitlab.com/ee/api/groups.html#:~:text=%22shared_runners_enabled%22%3Atrue%2C)   
    *
